### PR TITLE
Add test page

### PR DESCRIPTION
**Pull request by maxbeatty**
the most important page of all

https://github.com/maxbeatty/jsperf.com/blob/breakdown/_tpl/testPage.tpl

### CODE REVIEW

File: server/web/public.js [hunk 1]
```diff
@@ -15,6 +15,23 @@ exports.register = function(server, options, next) {
...
 
+  server.route({
+    method: "GET",
+    path: "/robots.txt",
+    handler: {
+      file: "public/robots.txt"
+    }
+  });
+
+  // TODO: should this be /public/ like non-precomposed?
```
Comment (maxbeatty) [inline server/web/public.js]: should this be `/public/apple-touch-icon-precomposed.png` like the non-precomposed icon? if you tell mobile safari where the one is, does it use that to try to find the other?

Comment (mathiasbynens) [inline server/web/public.js]: `precomposed` nowadays isn’t really needed anymore (was only for old iOS). Let’s just do `/apple-touch-icon.png`.

File: package.json [hunk 2]
```diff
@@ -38,10 +40,13 @@
     "handlebars": "^3.0.0",
     "hapi": "^8.2.0",
     "hapi-auth-cookie": "^3.0.1",
+    "highlight.js": "^8.6.0",
     "joi": "^5.1.0",
     "lodash": "^3.3.1",
+    "marked": "^0.3.3",
     "mysql": "^2.5.5",
     "prompt": "^0.2.14",
+    "ua-parser-js": "^0.7.9",
```
Comment (mathiasbynens) [inline package.json]: Until now, jsPerf was using https://github.com/bestiejs/platform.js for this. It’s available on npm. Any particular reason for the switch?

Comment (mathiasbynens) [inline package.json]: Until now, jsPerf was using https://github.com/bestiejs/platform.js for this (in the front-end). It’s available on npm; we could easily use it for the back-end as well. Or do you prefer ua-parser-js for some reason?

Comment (maxbeatty) [inline package.json]: I overlooked `platform.src.js` on the client. I was working backwards from [test page](https://github.com/maxbeatty/jsperf.com/pull/1/files#diff-3089e3bc030ce0d2deec583de5ec8723L97) to [`userAgent`](https://github.com/maxbeatty/jsperf.com/pull/1/files#diff-6a2d247a6d06eee315e37e7696833f79L94) to [browser.php](https://github.com/maxbeatty/jsperf.com/blob/master/_inc/browser.php) so just tried to find a suitable replacement. Looks like I can remove user agent on the server altogether

**Comment by jdalton** [comment]
platform.js is in maintenance mode so I wouldn't mind handing off to [uap-core](https://github.com/ua-parser/uap-core) by way of [node-uap](https://www.npmjs.com/package/node-uap).

**Comment by maxbeatty** [comment]
I'm about to be AFK for the next two weeks and was really hoping to finish this PR but alas there's still a few more tests to write :crying_cat_face: hopefully someone can fill in the rest or I can pick it up when I get back :smile_cat: 

```
$ npm run test-cov

> jsperf.com@2.0.0 test-cov /Users/max/Sites/jsperf.com
> lab -t 100 -c -L


  
  ..................................................
  ..................................................
  ..................................................
  ............

162 tests complete
Test duration: 2392 ms
No global variable leaks detected
Coverage: 99.52% (8/1653)
server/web/test/index.js missing coverage on line(s): 57, 58, 61, 62, 65, 66, 70, 82
Code coverage below threshold: 99.52 < 100
Linting results: No issues
```

**Comment by asilluron** [comment]
On it, thanks @maxbeatty

**Comment by asilluron** [comment]
Tagging in @mathiasbynens for a final review.

**Comment by mathiasbynens** [comment]
LGTM. Thanks, everyone!

**Pull request merged**